python/mxnet/gluon/block.py

```diff
@@ -1037,10 +1024,25 @@ def _build_cache(self, *args):
                          for name in out.list_arguments()]
             aux_array = [args[data_names[name]] if name in data_names.keys() else params[name].data()
                          for name in out.list_auxiliary_states()]
+
             # Partition the graph.
             out = out.optimize_for(self._backend, arg_array, aux_array, ctx, **self._backend_opts)
```

Comment:
I think we need to remove deleted params from the `params` dict too. Otherwise we just remove them from the `arg_array` and `aux_array` but not necessarily from the `params` or `self._params`

Reply:
Good catch, I added it in the latest commit